### DISCUSSION TITLE
Welcome to ft817_buddy Discussions!

### DISCUSSION THREAD

**Discussion by g7uhn** [g7uhn/ft817_buddy#21] [General]
## 👋 Welcome!
  We’re using Discussions as a place for feedback and suggestions on the FT-817 Buddy project. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others, be positive and are open-minded. Remember that this is a friendly community we
  build together 💪.

**Comment by g1bvi** [comment]
Hi Andy
Awaiting my pre soldered PCBs  from JLC and decided to order case etc.....
Discovered that my PCBs are 113 x 73  not 119 x 76 like yours ?

I uploaded Gerbes.zip direct to JCL and have re checked it.
The consequence is that it does not fit the 120 x 80 Hammond box neatly.

Its too late now but should I have edited the dimension to give a board edge?
73

Nick

**Reply by g7uhn** [reply]
Hi Nick,

Just checked mine.  113 x 73 mm is the correct Rev4 size that matches up to the Hammond 1591TC lid so you're good to go.

119 x 76 is the Rev3 size... there must be an old reference somewhere in the Docs that you've spotted... I think that was on page 1 of the Build Instructions where I've got a screenshot of me ordering my Rev3 boards but can you confirm?  I'll update this shortly so it doesn't scare anyone else.  You've got sharp eyes, great spot and thanks for giving me the heads up!

(Update: screenshots of JLCPCB ordering on Build Instructions now updated to match Rev4 :thumbsup:)

Great to hear you've got some boards on the way, can't wait to hear how you get on 😃 

Andy

**Comment by g1bvi** [comment]
Thanks Andy
The 1591TC  is 120 x 80 2 external dimensions.
Your pics showed board mounted through lid mounting holes.
Is there a UK source for the keyed chip I wonder?
Not great at CW to be honest but need to re learn it.

I am also going to use it with a 818 so will let you know how that goes.
Great little project.
Keep safe
73
Nick

**Reply by g7uhn** [reply]
Yep, that's correct, the Rev4 board matches the mounting holes on the 1591TC lid and the board dimensions are smaller than the outside edges of the lid to allow for a 3D-printed bottom section which is work-in-progress at this time.  [Some detail pics of how the PCB meets the lid are here](https://docs.google.com/document/d/1aEO0aWywQayS-OdA7QQsbSc2tqqowsrMStnaMw6trrY/edit?usp=sharing).

Very interested in how it goes with the FT-818.  I've heard conflicting things about whether or not the EEPROM set/get commands are available on it but not seen any confirmation either way.  If the EEPROM commands aren't available on the FT-818 there should still be plenty of use for the Buddy, just depends how you operate and what functions you want to use it for.

Cheers,

Andy

**Comment by g1bvi** [comment]
Found keyer chip from Kanga.
73
Nick

**Comment by ok2jiq** [comment]
Hello, does anyone have a rev 4 board for sale?  Or will someone not enter production? I would be interested in a plate with a described location in China. Jiri OK2JIQ 73!

**Reply by g7uhn** [reply]
Hi Jiri, I have a spare board here, I'll drop you an email to your QRZ.com listed address  :-)

**Reply by g7uhn** [reply]
...ah email bounced.  Drop me a line on my QRZ.com listed email.

**Reply by ok2jiq** [reply]
Hi Andy! I send email for you an your adress QRZ.com an hotmail.com  Thanks my email is ok2jiq@seznam.cz 
Sorry, I entered the wrong address on the qrz, I have controll and already corrected it. VY 73! de Jirka OK2JIQ

**Comment by oe9fwv** [comment]
hi Andy,
I built the 817 Buddy and it works fine, the only thing I changed in the code is that I put the ABC readout into the slow loop so it is displayed when I change it. 
One question I cannot answer, when I switch the narrow filter setting, the transceiver changes band, frequency and mode for half a second, switches back to the original mode and frequency to change the filter setting. Why? Can I change this somehow?

**Reply by g7uhn** [reply]
Hi Werner, great to hear you've built a working 817 Buddy, always makes me smile! :-D

It's been a while since I've used mine (a KX2 arrived in my life so that has become the primary portable radio) and memory of the code is hazy... 

The momentary freq/mode change is to do with the 817's two VFOs.  For some of the parameters (I have a feeling it's all the ones that are changed by EEPROM writing) we achieve this by temporarily switching to the VFO that isn't in use, toggle the value of the VFO that we've just switched away from, and then switch back the the original VFO, now with the changed parameter.  These functions are described in the ft817 library, not in the main block of Buddy code, here's a link: https://github.com/g7uhn/ft817_buddy/blob/c98d4895c375bed1803e98695699e904cbea380e/Arduino/lib/ft817/ft817.cpp#L123

Yes, it's a bit clunky, but it only takes maybe a second so I figured I could live with it.

I can't remember if the fundamental reason was that you cannot adjust the EEPROM parameters of the active VFO or if there is some risk in doing so (and therefore the low risk approach is to change the EEPROM parameters of the standby VFO).  Reading KA7OEI's full guide to the EEPROM interface will probably reveal the answer: http://www.ka7oei.com/ft817_meow.html 

Hope that helps,
Andy
